python/gink/impl/key_set.py

```diff
@@ -0,0 +1,222 @@
+        """ Returns a Boolean stating whether the key set is a subset of the specified set/list/tuple """
+        as_of = self._database.resolve_timestamp(as_of)
+        iterable = self._database.get_store().get_keyed_entries(
+            container=self.get_muid(), behavior=self.BEHAVIOR, as_of=as_of)
+        for entry_pair in iterable:
+            if entry_pair.builder.deletion:
+                continue
+            if decode_key(entry_pair.builder) not in superset:
+                return False
+        return True
+    
+    def isdisjoint(self, s: Iterable, *, as_of: GenericTimestamp=None) -> bool:
+        """ Returns a boolean stating whether the key set contents completely overlap with the specified set/list/tuple 
+            Sets are disjoint if and only if their intersection is an empty set.
+        """
+        return False if set(self.intersection(s, as_of=as_of)) else True
+    
+    def difference(self, s: Iterable, *, as_of: GenericTimestamp=None) -> Iterable:
+        """ Returns an iterable of keys in the key set that are not in the specified sets/lists/tuples  """
+        as_of = self._database.resolve_timestamp(as_of)
+        iterable = self._database.get_store().get_keyed_entries(
+            container=self.get_muid(), behavior=self.BEHAVIOR, as_of=as_of)
+        for entry_pair in iterable:
+            if entry_pair.builder.deletion:
+                continue
+            if decode_key(entry_pair.builder) not in s:
+                yield decode_key(entry_pair.builder)
+    
+    def intersection(self, s: Iterable, *, as_of: GenericTimestamp=None) -> Iterable:
+        """ Returns an iterable with elements common to the key set and the specified iterables """
+        for element in s:
+            if self.contains(element, as_of=as_of):
+                yield element
+    
+    def symmetric_difference(self, s: Iterable, *, as_of: GenericTimestamp=None) -> set:
+        """ Returns a new set with elements in either the key set or the specified iterable, but not both. """
+        elements = self.union(s, as_of=as_of)
+        for element in s:
+            if self.contains(element, as_of=as_of):
+                elements.remove(element)
+        return elements
+    
+    def union(self, s: Iterable, *, as_of: GenericTimestamp=None) -> set:
+        """ Returns a new set with elements from both the key set and the specified set """
+        return set(self.items(as_of=as_of)).union(s)
+    
+    def difference_update(self, s: Iterable, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Updates the key set, removing elements found in the specified iterables. """
+        for element in s:
+            if self.contains(element):
+                self.remove(element, bundler=bundler, comment=comment)
+        
+    def intersection_update(self, s: Iterable, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
```

Comment:
Same thing here whereby you'll want to ensure everything happens in a single bundle.